packages/core/src/rpc/substrate/state.ts

```diff
@@ -56,6 +56,24 @@ export const state_getKeysPaged: Handler<[string, number, string, HexString], st
   return block?.getKeysPaged({ prefix, pageSize, startKey })
 }
 
+/**
+ * Mock get proof of the runtime storage value.
+ *
+ * @param context
+ * @param params - [`keys`, `blockhash`]
+ *
+ * @return mock proof
+ */
+export const state_getReadProof: Handler<
+  [HexString[], HexString],
+  { at: HexString; proof: HexString[] } | null
+> = async (_context, [keys, hash]) => {
+  return {
+    at: hash,
+    proof: keys.map(() => ('0x' + '7'.repeat(64)) as HexString),
+  }
```

Comment:
Is there any way we could get a real proof here ? As far as I know, for this we don't need signing keys.

Reply:
Yes

Reply:
@serban300 I've implemented `state_getReadProof` to return proof of requested keys but the `trie_state_root` won't be the same as block `trie_state_root`. I don't know if correct hash is required and not sure if we can do that. Can you give this a try first before I look deeper into the implementation?

Reply:
@ermalkaleci thanks ! Yes, will try this. Just have to prioritize some reviews first, and then I will get to this. Hopefully this week.

Reply:
Checked. This doesn't work. I'm getting an error when validating the storage proof. We would need the `trie_state_root` to be the same as the block `trie_state_root`. Since I think that's how we're validating the proof. Would that be possible ?

Reply:
I'm not sure